From 41b7e79e0b322ae2262e4213b63137cfbcb4d3b7 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Sat, 24 Feb 2024 01:07:02 +0000
Subject: [PATCH] apps_cpp: edgeai_utils: Fix -Werror=format-security errors
 with mvprintw

To fix below building issue:
apps_cpp/common/src/edgeai_utils.cpp:276:17: error: format not a string literal and no format arguments [-Werror=format-security]
|   276 |         mvprintw(row++, 1, border.c_str());
|       |         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: pending

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 common/src/edgeai_utils.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/common/src/edgeai_utils.cpp b/common/src/edgeai_utils.cpp
index 264982b..1c86de8 100644
--- a/common/src/edgeai_utils.cpp
+++ b/common/src/edgeai_utils.cpp
@@ -273,19 +273,19 @@ void Statistics::reportingLoop(const string &demoName)
         clear();
         int row = 1;
 
-        mvprintw(row++, 1, border.c_str());
+        mvprintw(row++, 1, "%s", border.c_str());
         mvprintw(row++, 1, fmt.c_str(), demoName.c_str());
-        mvprintw(row++, 1, border.c_str());
+        mvprintw(row++, 1, "%s", border.c_str());
 
         for (uint64_t i = 0; i < statsDb.size(); i++)
         {
             auto const s = &statsDb[i];
 
-            mvprintw(row++, 1, border.c_str());
+            mvprintw(row++, 1, "%s", border.c_str());
             mvprintw(row++, 1, fmt.c_str(), s->m_inputName.c_str());
             mvprintw(row++, 1, fmt.c_str(), s->m_modelName.c_str());
             mvprintw(row++, 1, fmt.c_str(), s->m_modelType.c_str());
-            mvprintw(row++, 1, border.c_str());
+            mvprintw(row++, 1, "%s", border.c_str());
 
             for (auto &key : gStatKeys)
             {
@@ -304,7 +304,7 @@ void Statistics::reportingLoop(const string &demoName)
                             m->unit.c_str(), m->samples, len+2);
             }
 
-            mvprintw(row++, 1, border.c_str());
+            mvprintw(row++, 1, "%s", border.c_str());
         }
 
         refresh();
-- 
2.35.5

